services/server/server.go

```diff
@@ -98,6 +99,13 @@ func (s *Server) parseEnv(env []string) {
 			if err == nil {
 				s.envDisableTelemetry = b
 			}
+
+		case "QANAPI_DATA_RETENTION":
```

Comment:
It should be `DATA_RETENTION`. Use `time.ParseDuration`. Unlike `METRICS_RESOLUTION`, there suffix is required.